workflow/lib/shared/segment_cellpose.py

```diff
@@ -148,28 +152,86 @@ def prepare_cellpose(data, dapi_index, cyto_index, logscale=True, log_kwargs=dic
     return np.array([red, green, blue])
 
 
+def estimate_diameters(
+    data,
+    dapi_index,
+    cyto_index,
+    channels=[2, 3],  # Default channels for cell estimation
+    cyto_model="cyto3",
+    cellpose_kwargs=dict(flow_threshold=0.4, cellprob_threshold=0),
+    gpu=False,
+    logscale=True,
+):
+    """Estimate optimal cell diameter using Cellpose's diameter estimation.
+
+    Args:
+        data (numpy.ndarray): Multichannel image data
+        dapi_index (int): Index of DAPI channel
+        cyto_index (int): Index of cytoplasmic channel
+        channels (list): Channel indices for diameter estimation [cytoplasm, nuclei]
+        cyto_model (str): Cellpose model type to use
+        cellpose_kwargs (dict): Additional keyword arguments for Cellpose
+        gpu (bool): Whether to use GPU
+        logscale (bool): Whether to apply log scaling to image
+
+    Returns:
+        tuple: Estimated diameters for (nuclei, cells)
+    """
+    # Prepare RGB image
+    log_kwargs = cellpose_kwargs.pop(
+        "log_kwargs", dict()
+    )  # Extract log_kwargs from cellpose_kwargs
+    rgb = prepare_cellpose(
+        data, dapi_index, cyto_index, logscale, log_kwargs=log_kwargs
+    )
+
+    # Find optimal nuclei diameter
+    print("Estimating nuclei diameters...")
```

Comment:
I'm not a big fan of printing function information like this. Short term it is quick help for testing, but long term these start to add up and looking at the output of Snakemake run can become quite cluttered. We need to have a longer conversation about a logging system and the best way to show outputs (for devs and users), but for now just something to keep in mind.

Reply:
Fair. Maybe open an issue about this as well?